eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java

```diff
@@ -238,6 +244,65 @@ public void call(EncoderWrapper encodingCodec, DecoderWrapper decodingCodec) thr
         verifyAllPairs(codingAction, Applications.class, jsonCodes);
     }
 
+    @Test
+    public void testBatchRequestEncoding() throws Exception {
+        InstanceInfo instance = InstanceInfoGenerator.takeOne();
+        List<ReplicationInstance> replicationInstances = new ArrayList<>();
+        replicationInstances.add(new ReplicationInstance(
+                instance.getAppName(),
+                instance.getId(),
+                System.currentTimeMillis(),
+                null,
+                instance.getStatus().name(),
+                instance,
+                Action.Register
+        ));
+        final ReplicationList replicationList = new ReplicationList(replicationInstances);
+
+        Action2 codingAction = new Action2() {
+            @Override
+            public void call(EncoderWrapper encodingCodec, DecoderWrapper decodingCodec) throws IOException {
+                String encodedString = encodingCodec.encode(replicationList);
+
+                ReplicationList decodedValue = decodingCodec.decode(encodedString, ReplicationList.class);
+                assertThat(decodedValue.getReplicationList().size(), is(equalTo(1)));
+            }
+        };
+
+        // In replication channel we use JSON only
+        List<CodecWrapper> jsonCodes = Arrays.asList(
```

Comment:
The test covers all combinations:
Encoding ReplicationList using {JacksonJson,JacksonJson}
Encoding ReplicationList using {JacksonJson,LegacyJacksonJson}
Encoding ReplicationList using {LegacyJacksonJson,JacksonJson}
Encoding ReplicationList using {LegacyJacksonJson,LegacyJacksonJson}

Encoding ReplicationListResponse using {JacksonJson,JacksonJson}
Encoding ReplicationListResponse using {JacksonJson,LegacyJacksonJson}
Encoding ReplicationListResponse using {LegacyJacksonJson,JacksonJson}
Encoding ReplicationListResponse using {LegacyJacksonJson,LegacyJacksonJson}

On Tue, Oct 6, 2015 at 2:50 PM, David Liu <notifications@github.com> wrote:

> In
> eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java
> <https://github.com/Netflix/eureka/pull/674#discussion_r41327448>:
>
> > +                Action.Register
> > +        ));
> > +        final ReplicationList replicationList = new ReplicationList(replicationInstances);
> > +
> > +        Action2 codingAction = new Action2() {
> > +            @Override
> > +            public void call(EncoderWrapper encodingCodec, DecoderWrapper decodingCodec) throws IOException {
> > +                String encodedString = encodingCodec.encode(replicationList);
> > +
> > +                ReplicationList decodedValue = decodingCodec.decode(encodedString, ReplicationList.class);
> > +                assertThat(decodedValue.getReplicationList().size(), is(equalTo(1)));
> > +            }
> > +        };
> > +
> > +        // In replication channel we use JSON only
> > +        List<CodecWrapper> jsonCodes = Arrays.asList(
>
> what about the reverse (encoded with JacksonJson and decoded with Legacy)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/eureka/pull/674/files#r41327448>.
>